modules/Profile/containers/ProfileContainer/ProfileContainer.tsx

```diff
@@ -1,33 +1,51 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 
 import { ActionIcon, Button, Flex, SimpleGrid, Title, Loader } from "@mantine/core";
 import { notifications } from "@mantine/notifications";
 import { FaRegEdit } from "react-icons/fa";
 import { IoMdClose } from "react-icons/io";
 
 import Navbar from "@/modules/Dasboard/components/Navbar/Navbar";
+import { ACCESS_TOKEN_LOCAL_STORAGE_KEY } from "@/shared/constants/app.constants";
 import { useGetUserDetailsQuery } from "@/shared/redux/rtk-apis/users/users.api";
 
+import StudentProfile from "../../components/StudentProfile/StudentProfile";
 import TeacherProfile from "../../components/TeacherProfile/TeacherProfile";
 import { useStyles } from "./ProfileContainer.styles";
 
 const ProfileContainer = () => {
   const { classes } = useStyles();
-  const { data: user, isLoading, error } = useGetUserDetailsQuery();
+  const [isClient, setIsClient] = useState(false);
```

Comment:
What is this for?

Reply:
I have done it in another way. I am fetching the `currentUser` and then checking if the currentUser.userId exist.